diff --git a/Source/_gltf2usd/usd_material.py b/Source/_gltf2usd/usd_material.py
index 4ed537d..09404f2 100644
--- a/Source/_gltf2usd/usd_material.py
+++ b/Source/_gltf2usd/usd_material.py
@@ -180,8 +180,8 @@ class USDPreviewSurface(object):
             texture_shader = usd_uv_texture.get_shader()
             texture_shader.CreateOutput('rgb', Sdf.ValueTypeNames.Float3)
             texture_shader.CreateOutput('a', Sdf.ValueTypeNames.Float)
-            self._diffuse_color.ConnectToSource(texture_shader, 'rgb')
-            self._opacity.ConnectToSource(texture_shader, 'a')
+            self._diffuse_color.ConnectToSource(texture_shader.GetOutput('rgb'))
+            self._opacity.ConnectToSource(texture_shader.GetOutput('a'))
 
     def _set_pbr_specular_glossiness_specular(self, pbr_specular_glossiness):
         specular_glossiness_texture = pbr_specular_glossiness.get_specular_glossiness_texture()
@@ -256,7 +256,7 @@ class USDPreviewSurface(object):
 
             if AlphaMode(alpha_mode) == AlphaMode.BLEND:
                 print("Blend mode detected!")
-                self._opacity.ConnectToSource(texture_shader, 'a')
+                self._opacity.ConnectToSource(texture_shader.GetOutput('a'))
                 self._opacity.Set(defaultAlpha)
 
             if AlphaMode(alpha_mode) == AlphaMode.MASK:
@@ -286,7 +286,7 @@ class USDPreviewSurface(object):
             usd_uv_texture._fallback.Set(scale_factor)
             texture_shader = usd_uv_texture.get_shader()
             texture_shader.CreateOutput('r', Sdf.ValueTypeNames.Float)
-            self._metallic.ConnectToSource(texture_shader, 'r')
+            self._metallic.ConnectToSource(texture_shader.GetOutput('r'))
 
     def _set_pbr_roughness(self, pbr_metallic_roughness):
         metallic_roughness_texture = pbr_metallic_roughness.get_metallic_roughness_texture()
@@ -308,7 +308,7 @@ class USDPreviewSurface(object):
             usd_uv_texture._fallback.Set(scale_factor)
             texture_shader = usd_uv_texture.get_shader()
             texture_shader.CreateOutput('r', Sdf.ValueTypeNames.Float)
-            self._roughness.ConnectToSource(texture_shader, 'r')
+            self._roughness.ConnectToSource(texture_shader.GetOutput('r'))
 
     def export_to_stage(self, usd_material):
         """Converts a glTF material to a usd preview surface
