--- a/Source/_gltf2usd/usd_material.py
+++ b/Source/_gltf2usd/usd_material.py
@@ -247,7 +247,7 @@
         else:
             opacity_threshold = None
         texture_shader = self._create_texture_shader(color_texture_image, Usd.Shade.UsdUVTexture, color_scale)
-        self._diffuse_color.ConnectToSource(texture_shader, 'rgb')
+        self._diffuse_color.ConnectToSource(texture_shader.GetOutput('rgb'))
         if opacity_threshold is not None:
             self._opacity.Set(opacity_threshold)
         elif color_texture_has_alpha:
@@ -256,7 +256,7 @@
                 opacity_shader = self._create_texture_shader(color_texture_image, Usd.Shade.UsdUVTexture)
             else:
                 opacity_shader = texture_shader
-            self._opacity.ConnectToSource(opacity_shader, 'a')
+            self._opacity.ConnectToSource(opacity_shader.GetOutput('a'))
         else:
             self._opacity.Set(1.0)

@@ -272,13 +272,13 @@
         if normal_texture_image:
             normal_scale = gltf_material.normal_scale()
             texture_shader = self._create_texture_shader(normal_texture_image, Usd.Shade.UsdUVTexture, normal_scale)
-            self._normal.ConnectToSource(texture_shader, 'rgb')
+            self._normal.ConnectToSource(texture_shader.GetOutput('rgb'))

     def _set_pbr_metallic_roughness(self, gltf_material):
         if gltf_material.has_pbr_metallic_roughness():
             pbr_metallic_roughness = gltf_material.pbr_metallic_roughness()
             texture_shader = self._create_texture_shader(pbr_metallic_roughness.metallic_roughness_texture_image(), Usd.Shade.UsdUVTexture)
-            self._metallic.ConnectToSource(texture_shader, 'b')
-            self._roughness.ConnectToSource(texture_shader, 'g')
+            self._metallic.ConnectToSource(texture_shader.GetOutput('b'))
+            self._roughness.ConnectToSource(texture_shader.GetOutput('g'))

     def _set_khr_material_pbr_specular_glossiness(self, gltf_material):
